lib/cli-engine.js

```diff
@@ -300,18 +300,31 @@ function processFile(filename, configHelper, options) {
 
 /**
  * Returns result with warning by ignore settings
- * @param {string} filePath File path of checked code
- * @returns {Result} Result with single warning
+ * @param {string} filePath - File path of checked code
+ * @param {string} baseDir  - Absolute path of base directory
+ * @returns {Result}           Result with single warning
  * @private
  */
-function createIgnoreResult(filePath) {
+function createIgnoreResult(filePath, baseDir) {
+    var message;
+
+    if (/^\./.test(path.basename(filePath))) {
+        message = "Hidden file ignored by default.  Use '--ignore-pattern !.*' to override.";
```

Comment:
I am not sure we should use relative paths here, since we use absolute paths when reporting.

I think it would be better not to recommend to uningnore everything, but just the current file. Incidentally, you could then use a single message for all 3 cases: "File ignored by default. Use '--ignore-pattern \"!" + filePath +"\"' to overrride." (I also added necessary quotes around the pattern)

And maybe use `ignoredPaths.contains(filePath, "default")` instead of explicitly checking each ignore pattern?